portality/tasks/reporting.py

```diff
@@ -173,24 +173,8 @@ def count(self, prov):
         if not prov.action.startswith("status:"):
             return
 
-        role_precedence = ["associate_editor", "editor", "admin"]
-        best_role = None
-        for r in prov.roles:
-            try:
-                if best_role is None:
-                    best_role = r
-                if role_precedence.index(r) > role_precedence.index(best_role):
-                    best_role = r
-            except ValueError:
-                pass                            # The user has a role not in our precedence list (e.g. api) - ignore it.
-
-        countable = False
-        if best_role == "admin" and (prov.action == "status:accepted" or prov.action == "status:rejected"):
-            countable = True
-        elif best_role == "editor" and prov.action == "status:ready":
-            countable = True
-        elif best_role == "associate_editor" and prov.action == "status:completed":
-            countable = True
+        best_role = self._get_best_role(prov.roles)
+        countable = self._is_countable(prov, best_role)
```

Comment:
you'd be sending ```None``` through ```_is_countable()``` if the user doesn't have a best role. But a user should have a best role, right, since they're editing things. I guess this is a niche case.